File: core/src/com/abyssia/game/world/Chunk.java
package com.abyssia.game.world;


import com.badlogic.gdx.graphics.g2d.SpriteBatch;

public class Chunk {

    public static final int CHUNK_WIDTH = 16;
    public static final int CHUNK_HEIGHT = 16;

    protected TileMap tileMap;

    protected World world;

    private int x;
    private int y;

    public Chunk(){
        this.tileMap = new TileMap();
        this.tileMap.setChunk(this);
    }

    public TileMap getTileMap() {
        return tileMap;
    }

    public void setTileMap(TileMap tileMap) {
        this.tileMap = tileMap;
        tileMap.setChunk(this);
    }

    public void render(SpriteBatch batch){
        this.tileMap.render(batch);
    }

    public World getWorld() {
        return world;
    }

    public void setWorld(World world) {
        this.world = world;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }
}

File: client/WorldManagers/Chunk.cpp
#include "Chunk.h"



Chunk::Chunk(Position &position) : position(position)
{
	tileMap = make_shared<TileMap>(shared_from_this(), TILEMAP_HEIGHT, TILEMAP_WIDTH);
}

Chunk::Chunk()
{
}


Chunk::~Chunk()
{
}

void Chunk::render(sf::RenderTarget &element)
{
	tileMap->render(element);
}

shared_ptr<TileMap> Chunk::getTileMap() const
{
	return tileMap;
}

void Chunk::setTileMap(shared_ptr<TileMap> tileMap)
{
	this->tileMap = tileMap;
}

Position Chunk::getPosition() const
{
	return position;
}

void Chunk::setPosition(Position & position)
{
	this->position = position;
}

shared_ptr<World> Chunk::getWorld()
{
	return world.lock();
}

void Chunk::setWorld(weak_ptr<World> world)
{
	this->world = world;
}